### ISSUE TITLE
How to generate semantic words

### ISSUE THREAD

**Issue by gujiuxiang** [eriche2016/image_caption_with_semantic_attenion#4]
Hi, could you tell me how to generate the semantic words ?
For current cocotalk.h5 from neuraltalk2, there is no 'semantic_words' exits. Thanks

**Comment by y657250418** [comment]
@gujiuxiang Did you solve this problem? I met the same issue.
Looking forward to your reply.
Thanks.

**Comment by gujiuxiang** [comment]
You can use https://github.com/gujiuxiang/MIL.pytorch to generate the attributes, however, the predicted attributes contain a lot of noise. During training, you can use the GT attributes (remove those functional words, e.g., 'on', 'of', etc.), and you can achieve super performance, but during testing, you need to train a powerful MIL model for attribute detection, good luck!

**Comment by y657250418** [comment]
@gujiuxiang Thanks for your reply, and it does helps me a lot.
But when I use prepro_coco.py to generate the cocotalk.h5 and cocotalk.json file for training, I still find that the  snapshot_iter_240000.caffemodel is required. I can't find it anywhere. Do you know how to get this file?
Thanks.

**Comment by gujiuxiang** [comment]
You can have a look at https://github.com/s-gupta/visual-concepts

**Comment by y657250418** [comment]
@gujiuxiang Thanks a lot, it works for me.

**Comment by y657250418** [comment]
@gujiuxiang Sorry to bother you again. 
When I use the comand to train the model with regularization on attention weights:
      sh run_reg_att.sh
I get the next error message:

/home/xx/torch/install/bin/luajit: bad argument #3 to '?' (torch.*Tensor expected, got nil)
stack traceback:
	[C]: at 0x7f8239486240
	[C]: in function '__newindex'
	./misc_saver2_reg_atten_ws/LanguageModel.lua:520: in function 'forward'
	train_reg_on_att.lua:412: in function 'lossFun'
	train_reg_on_att.lua:479: in main chunk
	[C]: in function 'dofile'
	...huan/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x00405e90

./misc_saver2_reg_atten_ws/LanguageModel.lua:520 is:
        self.att_weights[{{}, {t, t}, {}}] = out[self.num_state+2]

I print the args :
 self.num_state:  2,
 out[self.num_state+2]:	nil	
 out:	{
  1 : CudaTensor - size: 240x512
  2 : CudaTensor - size: 240x512
  3 : CudaTensor - size: 240x9568
}
But I still can't understand why use out[self.num_state+2], and don't know how to solve it, can you help me?

**Comment by gorgeousyouth** [comment]
> You can have a look at https://github.com/s-gupta/visual-concepts

excuse me, if i want to use the groundtruth labels, how can i represent the label feature , only use glove embedding? and i cannot find the groundtruth label about each image about MSCOCO

**Comment by mymuli** [comment]
I wonder if anyone has succeeded in this experiment. If it succeeds, can you share the code? Thank you very much.
